client/my-sites/plugins/controller.js

```diff
@@ -74,7 +75,8 @@ function renderSinglePlugin( context, siteUrl, isWpcomPlugin ) {
 			isWpcomPlugin: isWpcomPlugin,
 			onPluginRefresh: title => titleActions.setTitle( title )
 		} ),
-		document.getElementById( 'primary' )
+		document.getElementById( 'primary' ),
+		context.store
```

Comment:
@dmsnell: This is NOT a react context - this is context from `page.js` and is provided explicitly by a route. It is accessible only via controllers

As a sidenote - Redux passes the store via context implicitly - `Provider` puts it there, and `connect` takes from it.

But we dont touch it explicitly at any point and we wont have tangled compoenents because of this, because:
- We dont touch `page` context.store - only `Provider` touches it, and more to the point - no component ever touches it. Only controllers get access to that.
- We dont touch `react` context.store - only `connect` touches it

And also, I'm pretty sure there's no other way to pass the store. I tried to put it in a module, but turns out it messes the SSR, @aduth pointed this out.

@johnHackworth:
> Maybe we should pass the subtrees that are going to be used by the component
I believe this is exactly the purpose of `connect`